libs/designer-ui/src/lib/picker/helpers.ts

```diff
@@ -26,6 +26,16 @@ export const filterAndSortItems = (
     if (!a.isParent && b.isParent) {
       return 1;
     }
-    return a.displayName.localeCompare(b.displayName);
+    return getTreeItemDisplayName(a).localeCompare(getTreeItemDisplayName(b));
   });
 };
+
+export const getTreeItemDisplayName = (item: TreeDynamicValue): string => {
+  const possibleValues = [item.displayName, item.value?.DisplayName, item.fullyQualifiedDisplayName];
```

Comment:
Why are we doing this? You should never assume the structure of data coming back from api call.... 

All dynamic tree calls be it legacy for swaggers or manifest ones they should be setting the displayName correctly because the connectors need to define at which path the display name is expected so a callback was added in the picker to fetch that.

Reply:
This is part of the problem with the fact that this code base has made such liberal use of `any`; we're left to make assumptions about the structure of data, especially because `TreeDynamicValue` has no top-level ID field.

If we want `TreeDynamicValue` to be a safe, strongly typed object, I can do that, but I didn't expect a change like that to be as well-received. Just let me know if that's preferred.

Reply:
If the problem is to put an Id to make each item unique we can do that, I dont have any objection to that.

My object is editor UI understanding the structure of the api data coming in - TreeDynamicValue needs to be strongly typed and the client services (connector service) needs to map the result from api response to the TreeDynamicValue structure.. And this is the only structure shared with UI editor, so it always knows Id -> Id, displayName -> displayName

Reply:
Fixed, moved `id` assignment to `ConnectorService`.